ts/sagas/wallet.ts

```diff
@@ -337,22 +397,57 @@ function* completionHandler(_: PaymentRequestCompletion) {
   yield put({ type: PAYMENT_COMPLETED });
 }
 
+function* fetchPagoPaToken(pagoPaClient: PagoPaClient): Iterator<Effect> {
+  const token: Option<string> = yield select(walletTokenSelector);
+  if (token.isSome()) {
+    const response:
+      | BasicResponseTypeWith401<SessionResponse>
+      | undefined = yield call(pagoPaClient.getSession, token.value);
+    if (response !== undefined && response.status === 200) {
+      // token fetched successfully, store it
+      yield put(storePagoPaToken(response.value.data.sessionToken));
+    }
+  }
+}
+
+function* watchWalletSaga(): Iterator<Effect> {
+  while (true) {
+    yield take(WALLET_TOKEN_LOAD_SUCCESS);
+
+    const pagoPaClient: PagoPaClient = PagoPaClient(pagoPaApiUrlPrefix);
+    yield put(storePagoPaToken("expired token"));
```

Comment:
perhaps you can use an Option here?

Reply:
Yep an `Option` does make sense here -- also, this line in particular was a test to check if the "retry with new session" logic was working -- I'll fix it right now